internal/datastore/mysql/tx.go

```diff
@@ -0,0 +1,29 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/authzed/spicedb/internal/datastore/mysql/migrations"
+)
+
+// BeginTXFunc is a polyfill for database/sql which does a closure style transaction lifecycle.
+// The underlying transaction is aborted if the supplied function returns an error.
+// The underlying transaction is committed if the supplied function returns nil.
+func BeginTxFunc(ctx context.Context, db *sql.DB, txOptions *sql.TxOptions, f func(*sql.Tx) error) error {
+	tx, err := db.BeginTx(ctx, txOptions)
+	if err != nil {
+		return err
+	}
+	defer migrations.LogOnError(ctx, tx.Rollback)
```

Comment:
Can we move this into the core mysql package? Its odd to rely on the migrations package

Reply:
This PR doesn't change anything about that. If you want to move it, file an issue or open another PR.